api/v1alpha1/shared_types.go

```diff
@@ -333,3 +339,22 @@ type KubernetesHorizontalPodAutoscalerSpec struct {
 	// +optional
 	Behavior *autoscalingv2.HorizontalPodAutoscalerBehavior `json:"behavior,omitempty"`
 }
+
+// MergeType defines the type of merge operation
+type MergeType string
+
+const (
+	// StrategicMerge indicates a strategic merge patch type
+	StrategicMerge MergeType = "StrategicMerge"
+	// JSONMerge indicates a JSON merge patch type
+	JSONMerge MergeType = "JSONMerge"
+)
+
+// KubernetesMergeSpec defines how to perform the merge operation
+type KubernetesMergeSpec struct {
+	// Type is the type of merge operation to perform
+	Type MergeType `json:"type"`
+
+	// Object contains the raw configuration for merged object
+	Object runtime.RawExtension `json:"object"`
```

Comment:
what the different with `apiextensionsv1.JSON`?

Reply:
Mostly the same, but rawextension has a wrapper about object for object converting.

But for mergepatch, apiextensionsv1.JSON is enough.